sphinx/util/docfields.py

```diff
@@ -298,7 +298,7 @@ def transform(self, node: nodes.field_list) -> None:
             # also support syntax like ``:param type name:``
             if typedesc.is_typed:
                 try:
-                    argtype, argname = fieldarg.split(None, 1)
+                    argtype, argname = fieldarg.rsplit(None, 1)
```

Comment:
@jakobandersen Does this change bring any regressions to C/C++ domains? This changes the split point of the `:param:` field.

* Before: `:param Dict[str, str] name:` -> `"Dict[str,"` and `"str] name"`
* After: `:param Dict[str, str] name:` -> `"Dict[str, str]"` and `"name"`

I think the identifiers of C/C++ are also not able to contain whitespace. So nothing affected.

Reply:
Right, this should only make things better. Identifier are indeed single words.